src/hooks/core/useCoreChatSystemHook.ts

```diff
@@ -274,41 +298,66 @@ const useCoreChatSystemHook = () => {
+          operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveChat,
+          id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+          task: async () => {
+            const chat = ChatSchema.fromWebsocketObject(websocketData);
+            await chat.save(localDb);
+          },
+          priority: QueueJobPriority.HIGH
+        });
+      }
+    }
+
+    if (Platform.OS === 'android') {
+      try {
+        websocketData?.originalMembers?.forEach(async (member) => {
+          const userMember = UserSchema.fromMemberWebsocketObject(
+            member,
+            websocketData?.channel?.id
+          );
+          await userMember.saveOrUpdateIfExists(localDb);
+        });
+      } catch (e) {
+        console.log('error on memberSchema');
+        console.log(e);
+      }
+
+      refresh('channelList');
+      refreshWithId('chat', websocketData?.channel?.id);
+      refresh('channelInfo');
+      refresh('channelMember');
+    } else {
       queue.addPriorityJob({
-        operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveChat,
-        id: `${websocketData?.channel?.id}-${websocketData?.message?.id}`,
+        operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveUserMember,
+        id: `${websocketData?.channel?.id}`,
+        priority: QueueJobPriority.HIGH,
         task: async () => {
-          const chat = ChatSchema.fromWebsocketObject(websocketData);
-          await chat.save(localDb);
-        },
-        priority: QueueJobPriority.HIGH
-      });
-    }
+          try {
+            websocketData?.originalMembers?.forEach(async (member) => {
+              const userMember = UserSchema.fromMemberWebsocketObject(
+                member,
+                websocketData?.channel?.id
+              );
+              await userMember.saveOrUpdateIfExists(localDb);
+            });
+          } catch (e) {
+            console.log('error on memberSchema');
+            console.log(e);
+          }
 
-    queue.addPriorityJob({
-      operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveUserMember,
-      id: `${websocketData?.channel?.id}`,
-      priority: QueueJobPriority.HIGH,
-      task: async () => {
-        try {
-          websocketData?.originalMembers?.forEach(async (member) => {
-            const userMember = UserSchema.fromMemberWebsocketObject(
-              member,
-              websocketData?.channel?.id
-            );
-            await userMember.saveOrUpdateIfExists(localDb);
-          });
-        } catch (e) {
-          console.log('error on memberSchema');
-          console.log(e);
+          refresh('channelList');
+          refreshWithId('chat', websocketData?.channel?.id);
+          refresh('channelInfo');
+          refresh('channelMember');
         }
-
-        refresh('channelList');
-        refreshWithId('chat', websocketData?.channel?.id);
-        refresh('channelInfo');
-        refresh('channelMember');
-      }
-    });
+      });
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This hunk introduces platform-specific logic for saving chat messages and user members. It's important to note that the `forEach` loop inside the `try` block is using an `async` function. This means that errors thrown inside this function will not be caught by the surrounding `catch` block because the function returns a promise. To fix this, consider using `Promise.all` with `map` instead of `forEach`.
```diff
+ if (Platform.OS === 'android') {
+   try {
+     await Promise.all(websocketData?.originalMembers?.map(async (member) => {
+       const userMember = UserSchema.fromMemberWebsocketObject(
+         member,
+         websocketData?.channel?.id
+       );
+       await userMember.saveOrUpdateIfExists(localDb);
+     }));
+   } catch (e) {
+     console.log('error on memberSchema');
+     console.log(e);
+   }
+ 
+   refresh('channelList');
+   refreshWithId('chat', websocketData?.channel?.id);
+   refresh('channelInfo');
+   refresh('channelMember');
+ } else {
+    queue.addPriorityJob({
+     operationLabel: DatabaseOperationLabel.CoreChatSystem_SaveUserMember,
+     id: `${websocketData?.channel?.id}`,
+     priority: QueueJobPriority.HIGH,
+      task: async () => {
+       try {
+         await Promise.all(websocketData?.originalMembers?.map(async (member) => {
+           const userMember = UserSchema.fromMemberWebsocketObject(
+             member,
+             websocketData?.channel?.id
+           );
+           await userMember.saveOrUpdateIfExists(localDb);
+         }));
+       } catch (e) {
+         console.log('error on memberSchema');
+         console.log(e);
+       }
+  
+       refresh('channelList');
+       refreshWithId('chat', websocketData?.channel?.id);
+       refresh('channelInfo');
+       refresh('channelMember');
+      }
+   });
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->